lib/worker.js

```diff
@@ -82,31 +77,16 @@ module.exports = function(id, task, done) {
       return done(err);
     }
 
-    // grab the last output
-    var output = outputs.pop();
-
-    // TODO tmp files aren't cleaned up when the process is terminated
-
     var cleanup = holdtime(function(err) {
       console.log("upload: %dms", arguments[arguments.length - 1]);
 
-      fs.unlink(output, function() {});
+      outputs.forEach(function(output) {
+        fs.unlink(output, function() {});
+      });
 
       return done(err);
     });
 
-    var uploader = new Uploader({
-      bucket: outputURI.hostname,
-      // strip the leading /
-      objectName: outputURI.pathname.slice(1),
-      objectParams: {
-        ACL: "public-read"
-      },
-      stream: fs.createReadStream(output)
-    });
-
-    uploader
-      .on("completed", cleanup)
-      .on("failed", cleanup);
+    upload(task.output, outputs, cleanup);
```

Comment:
In short: return values generally get ignored (except when working with streams), but using returns is useful for simplifying control flow.